### PR TITLE
Migrate to IntelliJ Platform V2

### PR DESCRIPTION
**Pull request by pderakhshanfar**
# Description of changes made
Update the project to migrate the plugin into the new platform version

# Why is merge request needed
To support new versions of IDEA

- [x] I have checked that I am merging into correct branch

### CODE REVIEW

**Review by arksap2002 — APPROVED** [review]

File: build.gradle.kts [hunk 1]
```diff
@@ -37,6 +42,10 @@ version = properties("pluginVersion")
 // Configure project's dependencies
 repositories {
     mavenCentral()
+    // this part is mandatory for all modules for platform version 2
+    intellijPlatform {
```
Comment (Vladislav0Art) [inline build.gradle.kts:46]: **nitpick**: a link to the appropriate documentation here would be nice to have.

File: build.gradle.kts [hunk 2]
```diff
@@ -169,23 +199,40 @@ dependencies {
...
+    publishing {
+        token = System.getenv("PUBLISH_TOKEN")
+        channels = listOf(properties("pluginVersion").split('-').getOrElse(1) { "default" }.split('.').first())
+    }
 
-    // Plugin Dependencies. Uses `platformPlugins` property from the gradle.properties file.
-    plugins.set(properties("platformPlugins").split(',').map(String::trim).filter(String::isNotEmpty))
+    pluginVerification {
+        ides {
+            recommended()
```
Comment (Vladislav0Art) [inline build.gradle.kts:223]: **nitpick:** A comment on what its function is would be nice.

File: src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt [hunk 3]
```diff
@@ -13,5 +13,5 @@ object LLMDefaultsBundle : DynamicBundle(LLMBundlePaths.defaults) {
      * Gets the requested default value.
      */
     @Nls
-    fun get(@PropertyKey(resourceBundle = LLMBundlePaths.defaults) key: String): String = getMessage(key)
+    fun get(@PropertyKey(resourceBundle = LLMBundlePaths.defaults) key: String): String = getMessage(key).split("|").last()
```
Comment (Vladislav0Art) [inline src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt:16]: This one should be commented; not trivial my splitting by "|" is needed.

**Review by Vladislav0Art — APPROVED** [review]

File: src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt (hunk 3 above)
Comment (pderakhshanfar) [reply src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt:16]: in the new version the Dynamic bundle returns the whole path + the value at the end. and each is separated by `|`. For example: `"|b|properties.llm.LLMDefaults|k|maxLLMRequest|3"` without this the plugin verification etc, will throw:
` java.lang.NumberFormatException: For input string: "|b|properties.llm.LLMDefaults|k|maxLLMRequest|3"`

File: gradle.properties
[the remark this answers is not in the document]
Comment (pderakhshanfar) [reply gradle.properties:1]: I usually do it later before release in a separate PR

File: src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt (hunk 3 above)
Comment (pderakhshanfar) [reply src/main/kotlin/org/jetbrains/research/testspark/bundles/llm/LLMDefaultsBundle.kt]: Added a comment explaining the purpose of this split

**Pull request merged**